pallets/permission0/src/ext/emission_impl.rs

```diff
@@ -369,3 +369,37 @@ pub fn toggle_permission_accumulation_impl<T: Config>(
 
     Ok(())
 }
+
+pub fn edit_stream_emission_permission<T: Config>(
+    origin: OriginFor<T>,
+    permission_id: PermissionId,
+    new_targets: BoundedBTreeMap<T::AccountId, u16, T::MaxTargetsPerPermission>,
+    new_streams: Option<BoundedBTreeMap<StreamId, Percent, T::MaxStreamsPerPermission>>,
+    new_distribution_control: Option<DistributionControl<T>>,
+) -> DispatchResult {
+    let who = ensure_signed(origin)?;
+
+    return Permissions::<T>::mutate(&permission_id, |permission| {
+        let Some(permission) = permission else {
+            return Err(Error::<T>::PermissionNotFound.into());
+        };
+
+        let current_block = frame_system::Pallet::<T>::block_number();
+
+        if permission.grantor == who {
+            match &permission.revocation {
+                RevocationTerms::RevocableByGrantor => {}
+                RevocationTerms::RevocableAfter(block) => {
+                    if &current_block <= block {
+                        return Err(Error::<T>::NotEditableYet.into());
+                    }
+                }
+                _ => return Err(Error::<T>::NotEditableYet.into()),
```

Comment:
Can we move this to a function in PermissionContract? PermissionContract::is_modifiable or something. `NotEditableYet` can become only `NotModifiable`.